app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -0,0 +1,148 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.util.BridgeCollectors.toImmutableList;
+
+import java.util.List;
+
+import javax.annotation.Resource;
+
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.SubpopulationDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.CriteriaUtils;
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifierImpl;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.FailedBatch;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+@Component
+public class DynamoSubpopulationDao implements SubpopulationDao {
+    
+    private DynamoDBMapper mapper;
+
+    /** DynamoDB mapper for the HealthDataRecord table. This is configured by Spring. */
+    @Resource(name = "subpopulationDdbMapper")
+    public final void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Override
+    public Subpopulation createSubpopulation(Subpopulation subpop) {
+        checkNotNull(subpop);
+        checkNotNull(subpop.getGuid());
+        checkNotNull(subpop.getStudyIdentifier());
+
+        if (subpop.getVersion() != null || subpop.isDeleted() || subpop.isRequired()) {
+            throw new BadRequestException("Subpopulation does not appear to be a new object");
+        }
```

Comment:
Added, also added test to verify that you can't set required or deleted on an update (so you can never set required or deleted when creating or updating. Required is internal and delete can only be changed through the delete call).